### PR TITLE
RouteTileDownloader: to kotlin

### PR DESCRIPTION
**Pull request by RingerJK**
## Description

Migrate Java to Kotlin `RouteTileDownloader` [#issue](https://github.com/mapbox/navigation-sdks/issues/167)

- [x] I have added any issue links
- [x] I have added all related labels (`bug`, `feature`, `new API(s)`, `SEMVER`, etc.)
- [x] I have added the appropriate milestone and project boards

## Testing

Please describe the manual tests that you ran to verify your changes

- [ ] I have tested locally (including `SNAPSHOT` upstream dependencies if needed) through testapp/demo app and run all activities to avoid regressions
- [ ] I have tested via a test drive, or a simulation/mock location app
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes

## Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have updated the `CHANGELOG` including this PR

### CODE REVIEW

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteTileDownloaderTest.kt [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+
+class RouteTileDownloaderTest {
+
+    @Test
+    fun startDownload_fetchRouteTilesIsCalled() {
+        val tilePath = "some/path/"
+        val offlineNavigator = mockk<OfflineNavigator>()
+        val listener = mockk<RouteTileDownloadListener>()
+        val offlineTiles = mockk<OfflineTiles>(relaxed = true)
+
```
Comment (Guardiola31337) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteTileDownloaderTest.kt]: Same as https://github.com/mapbox/mapbox-navigation-android/pull/2099#pullrequestreview-297786478 re: triple A / new lines

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteTileDownloaderTest.kt [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+    }
+
+    @Test
+    fun onError_downloadListenerErrorTriggered() {
+        val tilePath = "some/path/"
+        val offlineNavigator = mockk<OfflineNavigator>()
+        val listener = mockk<RouteTileDownloadListener>(relaxed = true)
+        val offlineError = mockk<OfflineError>()
+        val downloader = RouteTileDownloader(offlineNavigator, tilePath, listener)
+
```
Comment (Guardiola31337) [inline libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteTileDownloaderTest.kt]: Same as https://github.com/mapbox/mapbox-navigation-android/pull/2099#pullrequestreview-297786478 re: triple A / new lines

**Comment by RingerJK** [comment]
@Guardiola31337 PTAL

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteTileDownloaderTest.kt (hunk 2 above)
Comment (RingerJK) [reply libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteTileDownloaderTest.kt]: done

File: libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteTileDownloaderTest.kt (hunk 1 above)
Comment (RingerJK) [reply libandroid-navigation/src/test/java/com/mapbox/services/android/navigation/v5/navigation/RouteTileDownloaderTest.kt]: done

**Pull request merged**